### ISSUE TITLE
Add Ability to pass SourceJS CLI options via object

### ISSUE THREAD

**Issue by Alanoll** [sourcejs/Source#243]
#241 added the ability to use NodeJS's **require** to load SourceJS as part of a NodeJS script.

This issue will need to add the ability to pass in the SourceJS CLI options via plain object to the **startServer** method.

It should follow the same inheritance/override pattern as the current CLI ability.

1. Load Options via Options file
2. Load Options via CLI
3. Load Options via Options object
4. Override using **process.env**

**Comment by Alanoll** [comment]
I'm thinking something simple matching the SourceJS options file:

`
{
    hostname: '',
    port: ''
}
`

Use of commander should be inspected throughout SourceJS, to ensure passed Options are used where neccesary (Tests come to mind).

**Comment by Alanoll** [comment]
I don't think this issue should hold up the release of 0.6.0. I'm not sure of the full scope of changes, and they may be large enough that pushing to the next release makes sense.

**Comment by operatino** [comment]
Indeed, 0.6.0 just needs to be released already. But feel free to draft a PR whenever you're ready.

**Comment by Alanoll** [comment]
Just adding a note here, so I don't forget when I come back to this.

I think we should do what Norbert did in this [changeset](https://github.com/sourcejs/Source/pull/225/commits/781693ff42d8d80f2c762ce1049febc99b10a576) where he moved all the server logic into a **server.js** file. I can either wait for his PR to be accepted (post-0.6.0) or we just deal with the annoying merge.

**Comment by ndelangen** [comment]
We could cherry-pick that change into a new branch and submit a new PR?
